### DISCUSSION TITLE
Supply arguments to debug/launch button from statusbar

### DISCUSSION THREAD

**Discussion by s133314** [microsoft/vscode-cmake-tools#2407] [Q&A] [answered]
I would like that if I use the debug or launch button that some arguments are automatically given (those arguments will always be the same so I just need to set them once).

So I mean when using these buttons: 
![image](https://user-images.githubusercontent.com/66961749/156167685-4d08095e-32ab-4299-94bb-d49aa25774d7.png)

Now it essentially just runs `./ConjugatedMass` but I would like it to always standard be `./ConjugatedMass -c 1`

And an additional question if allowed. Now from the status bar you see `Release` is specified, if I now press the debug button will it automatically reconfigure and recompile with `Debug` before starting the debug?

**Comment by bobbrow** [comment, marked as answer]
You can use the `cmake.debugConfig` setting to add options to the debug and run buttons.  For example:
```
    "cmake.debugConfig": {
        "args": ["-c", "1"]
    }
```

> Now from the status bar you see `Release` is specified, if I now press the debug button will it automatically reconfigure and recompile with `Debug` before starting the debug?

Yes, it should.  If it does not, then we would consider that to be a bug in the extension.

**Reply by s133314** [reply]
Thanks it worked out. 

Just for the other newbies out here: you should set the `cmake.debugConfig` in `settings.json` in your `.vscode` folder.

**Reply by zethon** [reply]
@bobbrow Just curious, is there any plan to make this easier to find? 

This seems like a basic thing that is pretty easy and intuitive to do in most IDEs. But for VSCode I had to go hunting on Google and find my way here before I could figure out how to do it. It would be great to specify command lines per target and without having to edit a json file.

**Reply by zethon** [reply]
Silence... I had to come back to this thread to remember how to do this again for another project.

Why isn't this basic functionality more accessible?

**Reply by gcampbell-msft** [reply]
@zethon Thanks for re-pinging. What exactly are you hoping to be better for this workflow? We view the debug button to be a one-press action into debugging. To keep this, we need the configuration to be from within a json file. 

However, are you asking for better visibility into knowing about the `debugConfig` setting? Or are you asking for an interactive experience upon pressing the debug button?

**Reply by zethon** [reply]
In every other IDE I've used, finding the place to do things like set command line parameters or set environmental variables (to be set during debugging) is intuitive and easy to find. Even in the often-frustrating Xcode it's easy to find out where to do this. 

However in this case when I wanted to do either of these extremely common tasks when I'm debugging something, I had to go searching Google. I don't mind that I have to edit a text file, but some guidance would be nice and maybe a button? Like if I saw a gear-button next to the debug button, it'd be the first thing I'd click when looking how to do this.

**Reply by gcampbell-msft** [reply]
@zethon Ah, I understand, we'll transfer this to an issue and mark it as an enhancement request. Thanks for your feedback!
